### ISSUE TITLE
score_k overwritten with score function - even with score_k = None

### ISSUE THREAD

**Issue by RobertArbon** [markovmodel/PyEMMA#1192]
As per the title:

```
data = np.random.choice(np.arange(5), 50)
model =  MaximumLikelihoodMSM(lag=5, score_method='vamp2', score_k=2)
model.fit(data)
print('score_k = {}'.format(model.score_k))
model.score(data)
print('score_k = {}'.format(model.score_k))
```

gives: 

```
score_k = 2
score_k = 5
```

Here's the problem in `maximum_likelihood_msm.py`

```
def score(self, dtrajs, score_method=None, score_k=None):
...
        # reset estimator data if needed
        if score_method is not None:
            self.score_method = score_method
        if self.score_k is not None:
            self.score_k = score_k
```
should be: 

```
def score(self, dtrajs, score_method=None, score_k=None):
...
        # reset estimator data if needed
        if score_method is not None:
            self.score_method = score_method
        if score_k is not None:
            self.score_k = score_k
```

Again, I've corrected this in my own feature branch. Do you want a PR with test case?

**Comment by marscher** [comment]
Thanks again for this report! A test for this would actually be nice, because it indicates a not sufficient testing on this method.
Do you need a hint where to add this?

**Comment by RobertArbon** [comment]
yeah, some guidance on the kind of test you want would be good.

**Comment by marscher** [comment]
I'd guess it is easier for you, if you just push your feature branch and I pr the test for you there ;)

**Comment by RobertArbon** [comment]
done.

**Comment by marscher** [comment]
fixed by #1194
